### PR TITLE
Guest post: Rust for DSP algorithms

### PR DESCRIPTION
**Pull request by fouge**
Waiting for your comments and corrections. 
If there is a subject you want me to dive more into, let me know.

### CODE REVIEW

File: _posts/2020-01-29-rust-lib-dsp.md [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+
+Compiling and running that example takes milliseconds.
+
+The power of Rust programs used in a CLI[^5] environment, along with the ease to develop crates for different targets, is a strong advantage to use Rust. Coupled with its standard library, Rust is able to talk to machines and fetch data from servers quite easily, then pass those data into your algorithms and gives results in seconds.
+
+## Closing
+
+We have seen that Rust keeps its promises regarding efficiency and portability, making it a great choice to develop your future (or old) DSP algorithms. Getting rid of CMSIS-DSP in order to enjoy the productivity gains brought with Rust is to be considered. 
+
+I would love people who started developping DSP algorithms using Rust, for any target to join the discussion.
```
Comment (mnaberez) [inline _posts/2020-01-29-rust-lib-dsp.md]: ~developping~ developing

Awesome article!  :100:

Comment (fouge) [reply _posts/2020-01-29-rust-lib-dsp.md]: Thanks @mnaberez

**Comment by franc0is** [comment]
Hey @fouge, I've pushed some edits to your post. Let me know if they are acceptable to you. Some additional comments:
1. You do not mention the optimization level of your C code. It would be nice to compare apples to apples in that regard and use `-O3`.
2. It would be great to compare code size as well as performance, since this is equally as important as speed on embedded. Could you add it?
3. The portability argument is a bit weak. Had we written straight up C code (not CMSIS-DSP), it would have been equally portable. I think it's worth adding more details on why the *combo* of tooling and portability is nice.
4. One of the reasons folks like "matlab, then C" is that algorithm developers know matlab (and sometimes python), as they are data scientists. It would make sense to mention that, and say something about Rust's modern language constructs and how they could attract data scientists.

We'd love to publish your post tomorrow night (PST), so if you've got time to make a few edits before then it will make for an even stronger post.

Thanks!

**Comment by tyhoff** [comment]
Great topic choice! I really enjoyed the read, and I'm shocked how easy it was to get something up and running. 

I agree with @franc0is assessment on code size. I honestly haven't seen many arguments against Rust for it's speed, but the fact that to compile a simple program is 5-15x larger than a C program. In the M3/M4 embedded world, speed is *rarely* the concern. It's usually code space and RAM.

----
Do you have the example code sitting around on your computer that could be easily published? One thing we try to do is commit the example code under /examples so that it's easy for users to download and try it out themselves. If not, no worries, but maybe it's convenient. 

Finally, I pushed some simple edits as well. I get particular about consistent spelling and formatting, so those are the things I touched.

**Comment by fouge** [comment]
I added some Rust advantages against other methods to develop for DSP, but to my mind, its made too quickly. I could have done better but it's getting late here and I won't be able to work on it tomorrow (or late in the evening)

@tyhoff I do have the Rust and C code but I won't be able to publish them before tomorrow

You guys can edit as you wish.

**Pull request merged**